### PR TITLE
Use EF_NOSHADOW to control weapon shadow state

### LINKED ISSUE #831: Attachments of cloaked players cast shadows
### Build Info

d40ea92

### Description

When a player cloaks, their shadow is not rendered. However, the attached items such as pistol, grenades etc. keep casting their shadows even in the cloaked state, when they shouldn't.

### To Reproduce

* Join a local map
* Spawn a bot: `bot_add`
* Use `bot_mimic 1`
* Cloak yourself, so the bot also cloaks
* Observe the bot's shadow

### Expected behavior

In the cloaked state, the bot casts no shadow whatsoever.

### Actual behavior

In the cloaked state, the bot still casts its attachments' shadows.

### Operating System

- [ ] Windows
- [X] Linux

### Version/Distro

Linux Mint 21.3 Cinnamon

### Machine's CPU

AMD Ryzen 5 7600

### Machine's GPU

AMD RX 580

### GPU's driver

Mesa 23.2.1-1ubuntu3.1~22.04.2

### Build's compiler

gcc (Ubuntu 11.4.0-1ubuntu1~22.04)

### Additional context and Screenshots

nt_dimensions_test0002

### PR DESCRIPTION
**Pull request by AdamTadeusz**
## Description

Weapons that weren't active weren't getting data updates and so ShadowCastType() wasn't being run for them. This pr switches how we control whether shadows are drawn for weapons specifically to instead use the effect EF_NOSHADOW. I've tested bots toggling cloak on and off manually, bots running out of cloak, bots dropping weapons while cloaked, bots dying while cloaked and the shadow behavior for all of these seems correct.

One situation I can think of that I haven't tested where this might break would be a cloaked player who enabled cloak while not in pvs running into view of the player.

## Toolchain

- Windows MSVC VS2022
- Linux GCC Distro Native [Specify distro + GCC version]
- Linux GCC 10 Sniper 3.0

## Linked Issues

- fixes #831

### CODE REVIEW

**Review by Rainyan — APPROVED** [review]

**Review by Rainyan — APPROVED** [review]

File: mp/src/game/client/neo/c_neo_player.h [hunk 1]
```diff
@@ -168,6 +168,10 @@ class C_NEO_Player : public C_HL2MP_Player
 	void CheckVisionButtons();
 	void CheckLeanButtons();
 	void PlayCloakSound();
+public:
```
Comment (brysondev) [inline mp/src/game/client/neo/c_neo_player.h:171]: Q: Does this need to be public?

File: mp/src/game/server/neo/neo_player.h [hunk 2]
```diff
@@ -211,6 +211,10 @@ class CNEO_Player : public CHL2MP_Player
 	void CheckLeanButtons();
 	void PlayCloakSound();
 	void CloakFlash();
+public:
+	void SetCloakState(bool state);
```
Comment (brysondev) [inline mp/src/game/server/neo/neo_player.h:215]: Ditto

**Review by brysondev — APPROVED** [review]
lgtm, just wondering why the method is public

**Comment by AdamTadeusz** [comment]
> lgtm, just wondering why the method is public

I was going to initially use it in neo_gamerules but it turns out that's not necessary, ill fix that

**Review by brysondev — APPROVED** [review]

**Review by Rainyan — APPROVED** [review]

**Pull request merged**